spec/features/filters/echo_token_filter_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+describe 'Echo Token Filter' do
+  let(:token_response) {'<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/"><SOAP-ENV:Header/><SOAP-ENV:Body><SOAP-ENV:Fault><faultcode>SOAP-ENV:Client</faultcode><faultstring>Caught exception when getting NAMS AUID from Token Service.java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.launchpad.nasa.gov/icam/api/sm/v1/validate</faultstring><detail><ns4:AuthorizationFault xmlns:ns2="http://echo.nasa.gov/echo/v10/types" xmlns:ns3="http://echo.nasa.gov/echo/v10" xmlns:ns4="http://echo.nasa.gov/echo/v10/faults"><ns4:ErrorCode>NAMSAuidNotFoundForToken</ns4:ErrorCode><ns4:OpsMessage xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:nil="true"/><ns4:SystemMessage>Caught exception when getting NAMS AUID from Token Service.java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.launchpad.nasa.gov/icam/api/sm/v1/validate</ns4:SystemMessage><ns4:Timestamp>2020-02-07T14:32:04.509Z</ns4:Timestamp><ns4:ErrorInstanceId>534FF4C2-40E6-A0CB-7ED0-07BE7C7121FE</ns4:ErrorInstanceId><ns4:Token>ThisIsAToken</ns4:Token></ns4:AuthorizationFault></detail></SOAP-ENV:Fault></SOAP-ENV:Body></SOAP-ENV:Envelope>'}
+
+  let(:no_token_response) {'<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/"><SOAP-ENV:Header/><SOAP-ENV:Body><SOAP-ENV:Fault><faultcode>SOAP-ENV:Client</faultcode><faultstring>Caught exception when getting NAMS AUID from Token Service.java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.launchpad.nasa.gov/icam/api/sm/v1/validate</faultstring><detail><ns4:AuthorizationFault xmlns:ns2="http://echo.nasa.gov/echo/v10/types" xmlns:ns3="http://echo.nasa.gov/echo/v10" xmlns:ns4="http://echo.nasa.gov/echo/v10/faults"><ns4:ErrorCode>NAMSAuidNotFoundForToken</ns4:ErrorCode><ns4:OpsMessage xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:nil="true"/><ns4:SystemMessage>Caught exception when getting NAMS AUID from Token Service.java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.launchpad.nasa.gov/icam/api/sm/v1/validate</ns4:SystemMessage><ns4:Timestamp>2020-02-07T14:32:04.509Z</ns4:Timestamp><ns4:ErrorInstanceId>534FF4C2-40E6-A0CB-7ED0-07BE7C7121FE</ns4:ErrorInstanceId></ns4:AuthorizationFault></detail></SOAP-ENV:Fault></SOAP-ENV:Body></SOAP-ENV:Envelope>'}
+  
+  let(:echo_provider) { Echo::Provider.new('http://example.com', Rails.configuration.services['echo_soap']['services']) }
+  
+  
+  context 'When token is returned in SOAP Response' do
+    before do
+      allow_any_instance_of(Faraday::Connection).to receive(:post).and_return(Faraday::Response.new(body: token_response, status: 500))
+    end
+
+    it 'filters the token' do
```

Comment:
Should this succeed?  In this instance, shouldn't the response contain `<ns4:ErrorCode>NAMSAuidNotFoundFor**Token**</ns4:ErrorCode>` ?

Reply:
The code only removes tags named "Token" inside the AuthorizationFault section of the response.   The ErrorCode's value is not removed.  I could test that the full structure of the SOAP request is undisturbed, but that isn't really the point of the test.   

In fact,  the full test is to look at the log file and see what was written when this test completes.  That is what we will have to do to test it in production.

Reply:
My interpretation of line 17 is that it is supposed to be checking the string conversion of an echo response for the text 'Token'.  I do not have an intuitive sense of what the string conversion of an echo response is, but I would would expect it to contain most/all of the fields in the response.  Based on that assumption, the fact that this test passes struck me as odd, because I would expect at least one of the 'Token's in the response to match it.  
I threw a byebug in the test and discovered that `echo_provider.test_endpoint_connection("https://test.com", :non_token_response).to_s` evaluates to `"#<Echo::Response:0x00007fa4f4c5b4a8>"`.

Reply:
Note that the code in lines 40-44 of base.rb only affects the log message, not the SOAP response.  I used the test case to generate the log message to confirm that the logging did not include the <ns4:Token> tag.

I probably shouldn't have included the test case in my commit and I will remove it.

Reply:
Ah, I had overlooked that.

The alternative is that you could test the log message.  There are examples of that in spec/cmr/cmr_client_spec.rb